### PR TITLE
Make sure stuff that couldn't be parsed actually prints an error

### PR DESCRIPTION
**Pull request by x3ro**
Lets take the following example:

```
IFE [B+J], 0
```

Trying to compile this (non-sense example) currently results in asm.py silently failing
because there is a KeyError when writing to the file. This commit
prevents that by checking if we got valid instructions in o, x and y
and failing if we do not.

### CODE REVIEW

File: asm.py [hunk 1]
```diff
@@ -145,7 +145,11 @@ def report_error(filename, lineno, error):
                     program.append(int(datum[2:], 16))
                 else:
                     program.append(int(datum))
-        
+
+        if not (type(o) is int and type(x) is int and type(y) is int):
```
Comment (ghost) [inline asm.py]:
```
if not all(isinstance(i, int) for i in (o, x, y)):
```

Comment (jtauber) [inline asm.py]: if any(i is None for i in (o, x, y)):

Comment (x3ro) [inline asm.py]: @jtauber  `i is None` won't work, because in the mentioned case one of `(o, x, y)` can actually be a string.

**Comment by jtauber** [comment]
if strings are getting through that sounds like a bug in the regexes (unless labels are allowed)

**Comment by x3ro** [comment]
Yep, it actually seems to be a bug in the regexes, I just wanted to prevent `asm.py` from silently failing. When entering

```
IFE [B+J], 0
```

into http://dcpu.ru it results in the following error message:

```
(Fatal) Only sum of register and literal as value is allowed
```

I just wasn't sure how to fix this in the regexes, sorry for forgetting to mention it.

**Comment by andre-d** [comment]
Man, their error messages are nice

**Pull request closed without merging**